plugin-server/src/worker/vm/upgrades/utils/fetchEventsForInterval.ts

```diff
@@ -0,0 +1,93 @@
+import { DateTime } from 'luxon'
+
+import { Element, RawClickHouseEvent, TimestampFormat } from '../../../../types'
+import { DB } from '../../../../utils/db/db'
+import { parseRawClickHouseEvent } from '../../../../utils/event'
+import { castTimestampToClickhouseFormat } from '../../../../utils/utils'
+import { HistoricalExportEvent } from './utils'
+
+export interface RawElement extends Element {
+    $el_text?: string
+}
+
+export const fetchEventsForInterval = async (
+    db: DB,
+    teamId: number,
+    timestampLowerBound: Date,
+    offset: number,
+    eventsTimeInterval: number,
+    eventsPerRun: number
+): Promise<HistoricalExportEvent[]> => {
+    const timestampUpperBound = new Date(timestampLowerBound.getTime() + eventsTimeInterval)
+
+    const chTimestampLower = castTimestampToClickhouseFormat(
+        DateTime.fromISO(timestampLowerBound.toISOString()),
+        TimestampFormat.ClickHouseSecondPrecision
+    )
+    const chTimestampHigher = castTimestampToClickhouseFormat(
+        DateTime.fromISO(timestampUpperBound.toISOString()),
+        TimestampFormat.ClickHouseSecondPrecision
+    )
+
+    // :TODO: Adding tag messes up the return value?
+    const fetchEventsQuery = `
+    SELECT
+        event,
+        uuid,
+        team_id,
+        distinct_id,
+        properties,
+        timestamp,
+        created_at,
+        elements_chain
+    FROM events
+    WHERE team_id = ${teamId}
+    AND timestamp >= '${chTimestampLower}'
+    AND timestamp < '${chTimestampHigher}'
+    ORDER BY timestamp
+    LIMIT ${eventsPerRun}
+    OFFSET ${offset}`
+
+    const clickhouseFetchEventsResult = await db.clickhouseQuery<RawClickHouseEvent>(fetchEventsQuery)
+
+    return clickhouseFetchEventsResult.data.map(convertClickhouseEventToPluginEvent)
+}
+
+const convertClickhouseEventToPluginEvent = (event: RawClickHouseEvent): HistoricalExportEvent => {
+    const clickhouseEvent = parseRawClickHouseEvent(event)
+    if (clickhouseEvent.event === '$autocapture' && clickhouseEvent.elements_chain) {
+        clickhouseEvent.properties['$elements'] = convertDatabaseElementsToRawElements(clickhouseEvent.elements_chain)
+    }
+    const parsedEvent = {
+        uuid: clickhouseEvent.uuid,
+        team_id: clickhouseEvent.team_id,
+        distinct_id: clickhouseEvent.distinct_id,
+        properties: clickhouseEvent.properties,
+        timestamp: clickhouseEvent.timestamp.toISO(),
+        now: DateTime.now().toISO(),
```

Comment:
unrelated: this used to be the `now` stored with events. I think we currently keep if for compatibility but probably should just get rid of this at some point